### ISSUE TITLE
ALS 23.0.18 for macOS fails to launch on M1 CPU

### ISSUE THREAD

**Issue by berkeleynerd** [AdaCore/ada_language_server#1148]
I'm uncertain if this version fails on Intel Macs but it reliably fails on M1 and M2 Macs. The previous version 23.0.17 continues to work so a mitigation is to revert to the previous version.

**Comment by eliericha** [comment]
Hello @berkeleynerd ,

I was able to reproduce the issue in my environment and I'm working on fixing it.

The issue is that we recently started using `objcopy` from GNU binutils to strip debug symbols from the ada_language_server executable. While this works fine on Windows and Linux, on macOS it produces a corrupted executable that cannot launch. The solution is to use `dsymutil` and `strip` tools on macOS which are better suited for macOS executables.

We should be able to resolve this early next week.
